File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        
        string = ""
        MOD = 10 ** 9 + 7
        for i in range(1, n + 1):
            string += bin(i)[2:]
        return int(string, 2) % MOD
        

File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.go
func concatenatedBinary(n int) int {
    length, res, mod := 0, 0, 1000000007
    
    for i:=1; i<=n; i++ {
        if i & (i-1) == 0 {
            length++
        }
        res = (res << length | i) % mod
    }
    return res
}